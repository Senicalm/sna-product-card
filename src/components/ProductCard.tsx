import styles from "../styles/styles.module.css";

import { useProduct } from "../hooks/useProduct";
import React, { createContext, CSSProperties} from 'react';
import { ProductContextProps, Product, onChangeArgs, InitialValues, ProductCardHandlers } from '../interfaces/interfaces';

export const ProductContext = createContext({} as ProductContextProps)
const { Provider } = ProductContext;

export interface Props{
    // children?:ReactElement|ReactElement[],
    children:(args:ProductCardHandlers) => JSX.Element,
    product:Product,
    className?:string,
    style?:CSSProperties,
    onChange?:(args:onChangeArgs) => void,
    value?:number,
    initialValues?: InitialValues
}

export const ProductCard = ( {children, product, className, style, onChange, value, initialValues}:Props ) => {

    const { counter,maxCount, isMaxCountReached, isMinCountReached, increaseBy, reset } = useProduct( {onChange, product, value, initialValues} );
    
    return (
        <Provider value={{
            counter,
            maxCount,
            increaseBy,
            product
        }}>
            <div
                className={`${styles.productCard} ${className}`}
                style={style}
            >
                { children({
                    count:counter,
                    isMaxCountReached:isMaxCountReached,
                    isMinCountReached:isMinCountReached,
                    product,
                    increaseBy,
                    reset
                }) }
            </div>
        </Provider>
    )
}


