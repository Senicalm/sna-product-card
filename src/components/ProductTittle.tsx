import React, { CSSProperties, useContext } from 'react';
import { ProductContext } from './ProductCard';

import styles from "../styles/styles.module.css";

export interface Props{
    tittle?:string,
    className?:string,
    style?:CSSProperties
}
export const ProductTittle = ({ tittle,className,style }:Props) => {

    const { product } = useContext(ProductContext);
   
    return (
        <span
            className={`${styles.productDescription} ${className}`}
            style={style}
        >
            {tittle ? tittle : product.tittle}
        </span>
    )
}