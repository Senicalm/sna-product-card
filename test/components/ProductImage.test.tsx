
import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage.tsx',() =>{

    test('Debe de mostrar el componente correctamente con la imagen', () => { 

        const wrapper = renderer.create(<ProductImage img='https://s1.eestatic.com/2023/01/24/social/736187247_230385154_1706x960.jpg'/>);
        // console.log(wrapper.toJSON());
        expect(wrapper.toJSON()).toMatchSnapshot();
     });

     test('Debe de mostrar el componente con la imagen del producto', () => { 
         const wrapper = renderer.create(
             <ProductCard product={product2}>
                 {
                     () => (
                         <ProductImage />
                     )
                 }
             </ProductCard>
         )

         expect(wrapper.toJSON()).toMatchSnapshot();
      });

})