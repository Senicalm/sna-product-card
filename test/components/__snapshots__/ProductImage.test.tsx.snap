// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage.tsx Debe de mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage.tsx Debe de mostrar el componente correctamente con la imagen 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://s1.eestatic.com/2023/01/24/social/736187247_230385154_1706x960.jpg"
/>
`;
